File: backend/src/constants/place-zones.ts
/** Khu vực trong campus — dùng lọc địa điểm trên bản đồ. */
export const PLACE_ZONES = [
  { id: 'khu_a', label: 'Khu A' },
  { id: 'khu_b', label: 'Khu B' },
  { id: 'library', label: 'Thư viện' },
  { id: 'dorm', label: 'Ký túc xá' },
  { id: 'cafeteria', label: 'Căng tin' },
  { id: 'sports', label: 'Sân thể thao' },
  { id: 'faculty_it', label: 'Khoa CNTT' },
  { id: 'faculty_biz', label: 'Khoa Kinh tế' },
  { id: 'other', label: 'Khác' },
] as const;

export type PlaceZoneId = (typeof PLACE_ZONES)[number]['id'];

const ZONE_IDS = new Set<string>(PLACE_ZONES.map((z) => z.id));

export const isValidPlaceZone = (zone?: string): boolean =>
  zone != null && ZONE_IDS.has(zone);

File: mobile/lib/core/constants/place_zones.dart
/// Khu vực campus — đồng bộ với backend `place-zones.ts`.
class PlaceZone {
  const PlaceZone({required this.id, required this.label});

  final String id;
  final String label;
}

abstract final class PlaceZones {
  static const all = [
    PlaceZone(id: 'khu_a', label: 'Khu A'),
    PlaceZone(id: 'khu_b', label: 'Khu B'),
    PlaceZone(id: 'library', label: 'Thư viện'),
    PlaceZone(id: 'dorm', label: 'Ký túc xá'),
    PlaceZone(id: 'cafeteria', label: 'Căng tin'),
    PlaceZone(id: 'sports', label: 'Sân thể thao'),
    PlaceZone(id: 'faculty_it', label: 'Khoa CNTT'),
    PlaceZone(id: 'faculty_biz', label: 'Khoa Kinh tế'),
    PlaceZone(id: 'other', label: 'Khác'),
  ];

  static PlaceZone? byId(String? id) {
    if (id == null || id.isEmpty) return null;
    for (final z in all) {
      if (z.id == id) return z;
    }
    return null;
  }

  static String labelFor(String? id) => byId(id)?.label ?? 'Chưa phân khu';
}
